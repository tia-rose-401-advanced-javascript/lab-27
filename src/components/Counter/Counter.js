/* eslint-disable no-mixed-spaces-and-tabs */
/* eslint-disable no-tabs */
import React from 'react';

class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: 0,
    };
  }

	handleButtonClick = (e) => {
	  e.preventDefault();
	  this.setState({ counter: this.state.counter + 1 });
	};

	handleDecrementClick = (e) => {
	  e.preventDefault();
	  this.setState({ counter: this.state.counter - 1 });
	};

	render() {
	  // eslint-disable-next-line prefer-destructuring
	  const counter = this.state.counter;
	  let color;
	  const colorIncrement = { color: 'bisque' };
	  const colorDecrement = { color: 'lightpink' };
	  if (counter > 0) {
	    color = colorIncrement;
	  } else if (counter < 0) {
	    color = colorDecrement;
	  }


	  return (
			<div className='counter'>
				<h4 className='count' style={color}>{this.state.counter}</h4>
				<a href="*" className='increment' onClick={this.handleButtonClick}>
					Going up
				</a>
				<a href="*" className='decrement' onClick={this.handleDecrementClick}>
					Going down
				</a>
			</div>
	  );
	}
}

export default Counter;
