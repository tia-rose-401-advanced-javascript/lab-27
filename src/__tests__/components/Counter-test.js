import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../components/Counter/Counter';

describe('Counter component', () => {
  it('has an initial state', () => {
    // eslint-disable-next-line no-undef
    const app = shallow(<Counter />);

    expect(app.find('.count').text()).toBe('0');
  });

  describe('Counter component', () => {
    it('has an initial state', () => {
      // eslint-disable-next-line no-undef
      const app = shallow(<Counter />);
  
      expect(app.find('.count').text()).toBe('0');
    });
  
    it('can increment by 1', () => {
      // eslint-disable-next-line no-undef
      const wrapper = mount(<Counter/>);
  
      wrapper.find('.increment').simulate('click');
      expect(wrapper.state('counter')).toBe(1);
  
      wrapper.find('.increment').simulate('click');
      expect(wrapper.state('counter')).toBe(2);
    });
  
    it('can decrement by 1', () => {
      // eslint-disable-next-line no-undef
      const wrapper = mount(<Counter/>);
  
      wrapper.find('.decrement').simulate('click');
      expect(wrapper.state('counter')).toBe(-1);
  
      wrapper.find('.decrement').simulate('click');
      expect(wrapper.state('counter')).toBe(-2);
    });
  });
  
  describe('Counter snapshot test', () => {
    it('renders to DOM correctly', () => {
      const component = renderer.create(<Counter />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
