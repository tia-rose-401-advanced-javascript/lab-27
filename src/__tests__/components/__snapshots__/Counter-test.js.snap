// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot test renders to DOM correctly 1`] = `
<div
  className="counter"
>
  <h4
    className="count"
  >
    0
  </h4>
  <a
    className="increment"
    href="*"
    onClick={[Function]}
  >
    Going up
  </a>
  <a
    className="decrement"
    href="*"
    onClick={[Function]}
  >
    Going down
  </a>
</div>
`;
